### PR TITLE
feat: only fetch and consider matching contexts for subscribed contributors

### LINKED ISSUE #249: Do not take subscribed contributors into account
Since #233

### PR DESCRIPTION
**Pull request by JalilArfaoui**
closes #249

### CODE REVIEW

**Comment by JalilArfaoui** [comment]
@lutangar you’ve always been soooooo long to review 🙃 ;-)

**Comment by JalilArfaoui** [comment]
@lutangar 😘

File: android/app/src/main/java/com/dismoi/scout/accessibility/BackgroundService.kt [hunk 1]
```diff
@@ -175,18 +175,31 @@ class BackgroundService : AccessibilityService() {
...
+//        subscribedContributorsId.joinToString("&contributors[]=")
+//      }"
+
+    Log.d(TAG, "Fetching URL $matchingContextsEndpoint")
+
     val request = Request.Builder().url(matchingContextsEndpoint).build()
     val client = OkHttpClient()
 
-    client.newCall(request).enqueue(object: Callback {
+    client.newCall(request).enqueue(object : Callback {
```
Comment (lutangar) [inline android/app/src/main/java/com/dismoi/scout/accessibility/BackgroundService.kt:195]: Is there Kotlin CS Linter/Fixer?
This seems inconsistent with the rest of the file.

File: android/app/src/main/java/com/dismoi/scout/accessibility/BackgroundService.kt [hunk 2]
```diff
@@ -175,18 +175,31 @@ class BackgroundService : AccessibilityService() {
 
   // TODO To be moved to repository
   private fun fetchMatchingContexts() {
-    val matchingContextsEndpoint = "https://notices.bulles.fr/api/v3/matching-contexts" // TODO to be moved to env configuration
+    val prefs = getSharedPreferences("wit_player_shared_preferences", Context.MODE_PRIVATE)
+    val matchingContextsQueryString = prefs.getString("url", "")
+    val matchingContextsEndpoint: String = "https://notices.bulles.fr/api/v3/matching-contexts?$matchingContextsQueryString"
```
Comment (lutangar) [inline android/app/src/main/java/com/dismoi/scout/accessibility/BackgroundService.kt:180]: > TODO to be moved to env configuration
 
 :+1:

File: android/app/src/main/java/com/dismoi/scout/accessibility/BackgroundService.kt [hunk 3]
```diff
@@ -175,18 +175,31 @@ class BackgroundService : AccessibilityService() {
...
+
+    // TODO URL should be composed here
```
Comment (lutangar) [inline android/app/src/main/java/com/dismoi/scout/accessibility/BackgroundService.kt:182]: You're not adding contributor ids anymore? Thought it was the purpose of this PR.

**Review by lutangar — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/com/dismoi/scout/accessibility/BackgroundService.kt (hunk 3 above)
Comment (JalilArfaoui) [reply android/app/src/main/java/com/dismoi/scout/accessibility/BackgroundService.kt:182]: @lutangar it’s already in `matchingContextsQueryString` … The whole querystring with ids has already been saved in preferences on subscription 🙈 

The commented code is what I’d like to do once we just save contributors IDs :-)

File: android/app/src/main/java/com/dismoi/scout/accessibility/BackgroundService.kt (hunk 1 above)
Comment (JalilArfaoui) [reply android/app/src/main/java/com/dismoi/scout/accessibility/BackgroundService.kt:195]: I use Android Studio with Kotlin Style Guide …

I’ve just checked and in this particuliar syntax, the linter insist to have a space …

Found source here :

> Put a space before : in the following cases:    after the object keyword

https://kotlinlang.org/docs/coding-conventions.html#colon

File: android/app/src/main/java/com/dismoi/scout/accessibility/BackgroundService.kt (hunk 3 above)
Comment (JalilArfaoui) [reply android/app/src/main/java/com/dismoi/scout/accessibility/BackgroundService.kt:182]: Does it answer your question @lutangar ?

Comment (lutangar) [reply android/app/src/main/java/com/dismoi/scout/accessibility/BackgroundService.kt:182]: Let's say it does... I suppose now is not time to change this.

**Pull request merged**